src/host/getset.cpp

```diff
@@ -1028,36 +1036,96 @@ void ApiRoutines::GetLargestConsoleWindowSizeImpl(const SCREEN_INFORMATION& cont
-            const auto clipViewport = clip ? Viewport::FromInclusive(*clip) : bufferSize;
+            const auto clipViewport = clip ? Viewport::FromInclusive(*clip).Clamp(bufferSize) : bufferSize;
             const auto sourceViewport = Viewport::FromInclusive(source);
-            Viewport readViewport;
-            Viewport writtenViewport;
+            const auto fillViewport = sourceViewport.Clamp(clipViewport);
 
-            const auto w = std::max(0, sourceViewport.Width());
-            const auto h = std::max(0, sourceViewport.Height());
-            const auto a = static_cast<size_t>(w * h);
-            if (a == 0)
+            writer.BackupCursor();
+
+            if (gci.GetVtIo()->GetDeviceAttributes().test(Microsoft::Console::VirtualTerminal::DeviceAttribute::RectangularAreaOperations))
             {
-                return S_OK;
+                // This calculates just the positive offsets caused by out-of-bounds (OOB) source and target coordinates.
+                //
+                // If the source rectangle is OOB to the bottom-right, then the size of the rectangle that can
+                // be copied shrinks, but its origin stays the same. However, if the rectangle is OOB to the
+                // top-left then the origin of the to-be-copied rectangle will be offset by an inverse amount.
+                // Similarly, if the *target* rectangle is OOB to the bottom-right, its size shrinks while
+                // the origin stays the same, and if it's OOB to the top-left, then the origin is offset.
+                //
+                // In other words, this calculates the total offset that needs to be applied to the to-be-copied rectangle.
+                // Later down below we'll then clamp that rectangle which will cause its size to shrink as needed.
+                const til::point offset{
+                    std::max(0, -source.left) + std::max(0, clipViewport.Left() - target.x),
+                    std::max(0, -source.top) + std::max(0, clipViewport.Top() - target.y),
+                };
+
+                const auto copyTargetViewport = Viewport::FromDimensions(target + offset, sourceViewport.Dimensions()).Clamp(clipViewport);
+                const auto copySourceViewport = Viewport::FromDimensions(sourceViewport.Origin() + offset, copyTargetViewport.Dimensions()).Clamp(bufferSize);
+                const auto fills = Viewport::Subtract(fillViewport, copyTargetViewport);
+                std::wstring buf;
+
+                if (!fills.empty())
+                {
+                    Microsoft::Console::VirtualTerminal::VtIo::FormatAttributes(buf, TextAttribute{ fillAttribute });
+                }
+
+                if (copySourceViewport.IsValid() && copyTargetViewport.IsValid())
+                {
+                    // DECCRA: Copy Rectangular Area
+                    fmt::format_to(
+                        std::back_inserter(buf),
+                        FMT_COMPILE(L"\x1b[{};{};{};{};;{};{}$v"),

```

Comment:
```
CSI 9/11	Pts; Pls; Pbs; Prs; Pps;   Ptd; Pld; Ppd $ v
            area to be copied	       destination
```

The omitted one here is: 
**Pps** is the number of the page where the rectangular area is located. Default: Pps = 1.

and that makes sense